### ISSUE TITLE
Failure to find correct OpenSSL version on Ubuntu 22.04

### ISSUE THREAD

**Issue by sebschmi** [sfackler/rust-openssl#1748]
**Example**

Create a new binary crate with the following `Cargo.toml`:
```
[package]
name = "test-openssl-error"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
openssl = "0.10.44"
```

And the following `src/main.rs`:
```
fn main() {
    use openssl::sha;

    let mut hasher = sha::Sha256::new();

    hasher.update(b"Hello, ");
    hasher.update(b"world");

    let hash = hasher.finish();
    println!("Hashed \"Hello, world\" to {:?}", hash);
}
```

Then run `cargo run`. After building, the program terminates with the following, before producing any output. Running the binary directly via `target/debug/test-openssl-error` produces the same error.
```
target/debug/test-openssl-error: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```

My system includes OpenSSL 3, but not 1.1. Somehow, it still selects 1.1 when building. Is there a way to debug why that is?

**OS info**:
```
$ lsb_release -a
LSB Version:	core-11.1.0ubuntu4-noarch:printing-11.1.0ubuntu4-noarch:security-11.1.0ubuntu4-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```

```
$ ldd target/debug/test-openssl-error
	linux-vdso.so.1 (0x00007fff87ab6000)
	libssl.so.1.1 => not found
	libcrypto.so.1.1 => not found
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f27aaff0000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f27aadc8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f27ab0a7000)
```

```
$ ldd ~/.cargo/bin/cargo
	linux-vdso.so.1 (0x00007ffe07de1000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f1205460000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f120545b000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f1205456000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f120536f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f120536a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f12043d8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f12054c5000)
```

**Related**:
The same seems to happen on arch: https://github.com/pksunkara/cargo-workspaces/issues/81

**Comment by sfackler** [comment]
It sounds like you have two OpenSSL installations on the system at the same time. I'm not sure what the easiest way to resolve that would be, but you could e.g. search the filesystem for all copies of `opensslv.h`.

**Comment by sebschmi** [comment]
Thanks, I found it. I have various versions of OpenSSL on my system because I use conda. Apparently the main conda environment was not set up correctly. I had to add it to the `LD_LIBRARY_PATH`, and now everything works as expected.

There must be a bug somewhere, since the version 1.1 is correctly found during build, but not during execution. But it is probably not in this crate. Thanks for the help!
